lib/common/watchdog.c

```diff
@@ -227,13 +227,22 @@ pcmk__locate_sbd(void)
     return sbd_pid;
 }
 
-long
+guint
 pcmk__get_sbd_watchdog_timeout(void)
 {
-    static long sbd_timeout = -2;
+    static guint sbd_timeout = 0;
+    static bool have_timeout = false;
 
-    if (sbd_timeout == -2) {
-        sbd_timeout = crm_get_msec(getenv("SBD_WATCHDOG_TIMEOUT"));
+    if (have_timeout == false) {
+        char *env = getenv("SBD_WATCHDOG_TIMEOUT");
+
+        if ((env != NULL) && (env[0] == 'P')) {
+            crm_warn("SBD-daemon doesn't allow '%s' for "
+                     "SBD_WATCHDOG_TIMEOUT", env);
```

Comment:
mention that the value is ignored

Reply:
Hmm ... maybe ignoring isn't the best thing to do here.
Thus maybe better to stay with the warning and pass explanation and exit to the caller.